dom_task(temp)/postList.js

```diff
@@ -0,0 +1,598 @@
+class PostList {
+    _posts = [];
+
+    constructor(posts) {
+        this._posts = posts.concat();
+    }
+
+    getPage(skip = 0, top = 10, filterConfig = undefined) {
+        if (typeof skip !== 'number' || typeof top !== 'number') {
+            console.log('Error with inputting types!');
+            return;
+        }
+
+        if (filterConfig){
+            let returningPosts = this._posts;
+
+            for (let param in filterConfig){
+                if (param === 'hashTags'){
+                    for (let i = 0; i < filterConfig.hashTags.length; i++){
+                        returningPosts = returningPosts.filter(post => post.hashTags.includes(filterConfig.hashTags[i]));
+                    }
+                }
+                else if (param === 'dateFrom'){
+                    returningPosts = returningPosts.filter(post => post.createdAt >= filterConfig.dateFrom);
+                }
+                else if (param === 'dateTo'){
+                    returningPosts = returningPosts.filter(post => post.createdAt < filterConfig.dateTo);
+                }
+                else if (param === 'author'){
+                    returningPosts = returningPosts.filter(post => post.author === filterConfig.author);
+                }
```

Comment:
Minor. ```switch...case``` will look better

Reply:
I fixed this, but don't know why it doesn't write outdated